src/renderers/shared/fiber/ReactFiberScheduler.js

```
@@ -1188,15 +1189,32 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       if (capturedErrors === null) {
         capturedErrors = new Map();
       }
-      capturedErrors.set(boundary, {
-        componentName,
-        componentStack,
-        error,
-        errorBoundary: errorBoundaryFound ? boundary.stateNode : null,
-        errorBoundaryFound,
-        errorBoundaryName,
-        willRetry,
-      });
+
+      if (__DEV__) {
+        capturedErrors.set(boundary, {
+          componentName,
+          componentStack,
+          error,
+          errorBoundary: errorBoundaryFound ? boundary.stateNode : null,
+          errorBoundaryFound,
+          errorBoundaryName,
+          willRetry,
+          shouldIgnoreErrorMessage: error != null &&
+            typeof error.__reactShouldIgnoreErrorMessage === 'boolean' &&
+            error.__reactShouldIgnoreErrorMessage,
+        });
+      } else {
+        capturedErrors.set(boundary, {
+          componentName,
+          componentStack,
+          error,
+          errorBoundary: errorBoundaryFound ? boundary.stateNode : null,
+          errorBoundaryFound,
+          errorBoundaryName,
+          willRetry,
+          shouldIgnoreErrorMessage: false,
+        });
+      }
```

I want it to be DCE'd, but if we change this to always ignore errors in DEV then this doesn't matter. https://github.com/facebook/react/pull/10353#issuecomment-319718373